docs/proposals/project-user-resource-quotas.md

```diff
@@ -0,0 +1,209 @@
+# Project resource quotas
+
+**Authors**: Lovro Sviben (@lsviben)
+
+**Status**: Draft proposal.
+
+**Issue**: https://github.com/kubermatic/kubermatic/issues/8042
+
+## Goals
+The idea is to allow admins to control how many resources are projects using by introducing resource quotas.
+
+Resources that will be under quota:
+- vCPU
+- RAM
+- storage (node disk size) - TBD, we can remove this as it's not a strict requirement
+
+## Non-Goals
+- The quotas won't work for imported external clusters.
+- The quotas won't restrict the usage of resources for the control plane on the seed cluster.
+- Nodes which join the cluster using other means then through KKP are not supported in the quotas.
+
+## Motivation and Background
+
+Clients would like to have options to fine-grain restrict resource usage per project.
+The reasons vary from being able to stop the depletion of resources in a shared datacenter, to being able to sell packages (projects) with a set 
+number of resources available. 
+
+This will be an EE feature.
+
+## How will the resource quotas work
+
+Project quota - amount of resources (vCPU, RAM, storage) that can allocated for nodes in a project.
+
+Rules:
+
+- The quotas are for node resources
+- The quotas are enforced(we should block creation of new nodes that would exceed the quota)
+- The resource quotas management will be admin-only.
+- By default, projects will not have a quota, meaning they can use as many resources they want
+- There will be a way to set a default resource quota for all projects
+- When a node is deleted, the quota usage needs to decrease for the project
```

Comment:
Do we want to do this once the Node is gone or once the Machine object is gone? In general, the Machine object should disappear after the node has gone, IIRC.

Reply:
Makes sense to do it when the Machine object is gone